aiohttp/web.py

```diff
@@ -332,140 +332,176 @@ def raise_graceful_exit():
+        if access_log_format is not None:
+            make_handler_kwargs['access_log_format'] = access_log_format
+        self._handler = app.make_handler(loop=loop, access_log=access_log,
+                                         **make_handler_kwargs)
+
+        self.shutdown_timeout = shutdown_timeout
+        self.scheme = scheme = 'https' if ssl_context else 'http'
+        self.ssl_context = ssl_context
+        self.backlog = backlog
+        self.access_log_format = access_log_format
+        self.access_log = access_log
+        self.base_url = URL('{}://localhost'.format(scheme)).with_port(port)
+
+        if path is None:
+            self.paths = ()
+        elif isinstance(path, (str, bytes, bytearray, memoryview))\
+                or not isinstance(path, Iterable):
+            self.paths = (path,)
+        else:
+            self.paths = path
 
-    scheme = 'https' if ssl_context else 'http'
-    base_url = URL.build(scheme=scheme, host='localhost', port=port)
+        if sock is None:
+            self.socks = ()
+        elif not isinstance(sock, Iterable):
+            self.socks = (sock,)
+        else:
+            self.socks = sock
 
-    if path is None:
-        paths = ()
-    elif isinstance(path, (str, bytes, bytearray, memoryview))\
-            or not isinstance(path, Iterable):
-        paths = (path,)
-    else:
-        paths = path
+        if host is None:
+            if (self.paths or self.socks) and not port:
+                self.hosts = ()
+            else:
+                self.hosts = ("0.0.0.0",)
+        elif isinstance(host, (str, bytes, bytearray, memoryview))\
+                or not isinstance(host, Iterable):
+            self.hosts = (host,)
+        else:
+            self.hosts = host
 
-    if sock is None:
-        socks = ()
-    elif not isinstance(sock, Iterable):
-        socks = (sock,)
-    else:
-        socks = sock
+        if self.hosts and port is None:
+            self.port = 8443 if ssl_context else 8080
 
-    if host is None:
-        if (paths or socks) and not port:
-            hosts = ()
-        else:
-            hosts = ("0.0.0.0",)
-    elif isinstance(host, (str, bytes, bytearray, memoryview))\
-            or not isinstance(host, Iterable):
-        hosts = (host,)
-    else:
-        hosts = host
-
-    if hosts and port is None:
-        port = 8443 if ssl_context else 8080
-
-    server_creations = []
-    uris = [str(base_url.with_host(host).with_port(port)) for host in hosts]
-    if hosts:
-        # Multiple hosts bound to same server is available in most loop
-        # implementations, but only send multiple if we have multiple.
-        host_binding = hosts[0] if len(hosts) == 1 else hosts
-        server_creations.append(
-            loop.create_server(
-                handler, host_binding, port, ssl=ssl_context, backlog=backlog
+        self._apps = []
+        self._servers = None
+
+    def register(self, app, *, prefix="/"):
```

Comment:
How could I unregister specific app in runtime?

Reply:
You can't at the moment. It's not meant to be mutable. I can try to implement it but this is not my priority. The code to create a server and register apps is pretty small and I suppose it's best to just re-instantiate a new one if you need to change something.

Reply:
Re-instantiate a new server means to stop all the applications and cause a downtime. Not a good behaviour for adding a new app.

Reply:
(Same comment: it wasn't meant to avoid downtime, it was meant to allow running the apps without having to make aiohttp the center of the process)